### ISSUE TITLE
ipex-xpu "OSError: [WinError 126] The specified module could not be found." on windows i7-1185G7 

### ISSUE THREAD

**Issue by KiwiHana** [intel/intel-extension-for-pytorch#429]
### Describe the bug

refer to this guide to install Ipex on windows10 i7-1185G7: https://intel.github.io/intel-extension-for-pytorch/xpu/latest/tutorials/installations/windows.html

VS 2022
oneAPI 2023.2
python 3.9 Miniconda

```
conda create -n ipex python=3.9
conda install pkg-config libuv
python -m pip install torch==2.0.0a0 intel_extension_for_pytorch==2.0.110+gitba7f6c1 -f https://developer.intel.com/ipex-whl-stable-xpu
```

```
> python -c "import torch; import intel_extension_for_pytorch as ipex; print(torch.__version__); print(ipex.__version__); [print(f'[{i}]: {torch.xpu.get_device_properties(i)}') for i in range(torch.xpu.device_count())];"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\hua1\Miniconda3\envs\ipex-xpu\lib\site-packages\intel_extension_for_pytorch\__init__.py", line 89, in <module>
    raise err
OSError: [WinError 126] The specified module could not be found. Error loading "C:\Users\hua\Miniconda3\envs\ipex-xpu\lib\site-packages\intel_extension_for_pytorch\bin\intel-ext-pt-gpu.dll" or one of its dependencies.
```

### Versions

VS 2022
oneAPI 2023.2
python 3.9 Miniconda

**Comment by jingxu10** [comment]
Have you got the oneAPI env activated?

**Comment by KiwiHana** [comment]
> Have you got the oneAPI env activated?
Yes， I have. The device has iGPU no Arc DGPU.

(ipex-pip) PS C:\Program Files (x86)\Intel\oneAPI> .\setvars.bat
:: initializing oneAPI environment...
   Initializing Visual Studio command-line environment...
   Visual Studio version 17.3.6 environment configured.
   "C:\Program Files\Microsoft Visual Studio\2022\Community\"
   Visual Studio command-line environment initialized for: 'x64'
:  compiler -- latest
:  dal -- latest
:  debugger -- latest
:  dev-utilities -- latest
:  dpcpp-ct -- latest
:  dpl -- latest
:  ipp -- latest
:  ippcp -- latest
:  mkl -- latest
:  tbb -- latest
:: oneAPI environment initialized ::
(ipex-pip) PS C:\Program Files (x86)\Intel\oneAPI> python -c "import torch; import intel_extension_for_pytorch as ipex; print(torch.__version__); print(ipex.__version__); [print(f'[{i}]: {torch.xpu.get_device_properties(i)}') for i in range(torch.xpu.device_count())];"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\huangya1\Miniconda3\envs\ipex-pip\lib\site-packages\intel_extension_for_pytorch\__init__.py", line 89, in <module>
    raise err
OSError: [WinError 126] The specified module could not be found. Error loading "C:\Users\huangya1\Miniconda3\envs\ipex-pip\lib\site-packages\intel_extension_for_pytorch\bin\intel-ext-pt-gpu.dll" or one of its dependencies.

(ipex-pip) PS C:\Program Files (x86)\Intel\oneAPI> pip list
Package                     Version
--------------------------- ------------------
filelock                    3.12.3
intel-extension-for-pytorch 2.0.110+gitba7f6c1
Jinja2                      3.1.2
MarkupSafe                  2.1.3
mpmath                      1.3.0
networkx                    3.1
numpy                       1.25.2
pip                         23.2.1
psutil                      5.9.5
setuptools                  68.0.0
sympy                       1.12
torch                       2.0.0a0+gitc6a572f
typing_extensions           4.7.1
wheel                       0.38.4

**Comment by jingxu10** [comment]
@min-jean-cho

**Comment by LeonNerd** [comment]
I have the same problem with windows,

**Comment by midnqp** [comment]
I have the same error but for `backend_with_compiler.dll` on Windows 11 with Intel Core i7 and Intel Irish Xe Graphics. I don't have CUDA or NVIDIA graphics card.  I followed the same installation guide.

```
OSError: [WinError 126] The specified module could not be found. Error loading "C:\Users\Muhammad\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\lib\backend_with_compiler.dll" or one of its dependencies.
```

**Comment by KiwiHana** [comment]
@bouachalazhar  You can install OneAPI 2024.0 and ipex 2.1, may work.
I have tried on MTL.

**Comment by doucej** [comment]
I have the same problem as @midnqp.

PS C:\Program Files (x86)\Intel\oneAPI> .\setvars.bat
:: initializing oneAPI environment...
   Initializing Visual Studio command-line environment...
   Visual Studio version 17.6.2 environment configured.
   "C:\Program Files\Microsoft Visual Studio\2022\Community\"
   Visual Studio command-line environment initialized for: 'x64'
:  advisor -- latest
:  compiler -- latest
:  dal -- latest
:  debugger -- latest
:  dev-utilities -- latest
:  dnnl -- latest
:  dpcpp-ct -- latest
:  dpl -- latest
:  ipp -- latest
:  ippcp -- latest
:  mkl -- latest
:  tbb -- latest
:  vtune -- latest
:: oneAPI environment initialized ::
PS C:\Program Files (x86)\Intel\oneAPI> python -c "import torch; import intel_extension_for_pytorch as ipex; print(torch.version); print(ipex.version); [print(f'[{i}]: {torch.xpu.get_device_properties(i)}') for i in range(torch.xpu.device_count())];"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\jdouc\AppData\Roaming\Python\Python310\site-packages\torch\__init__.py", line 139, in <module>
    raise err
OSError: [WinError 126] The specified module could not be found. Error loading "C:\Users\jdouc\AppData\Roaming\Python\Python310\site-packages\torch\lib\backend_with_compiler.dll" or one of its dependencies.
PS C:\Program Files (x86)\Intel\oneAPI>

Installed oneAPI 2024.0. Tried miniconda installation too, similar issues there. Something not setting up library search path correctly?

Trying intel-extension-for-pytorch 2.1.10+xpu here. Running on Win11 with an Arc A750.

**Comment by yinghu5** [comment]
@doucej,  Hi  please see #485. if possible, please use the anaconda prompt itself, not in PS (powershell) environment. 

Neither “windows powershell” or “anaconda powershell prompt” does not work, if using conda prompt, we need to use "anaconda prompt".

**Comment by yinghu5** [comment]
Dear @doucej @midnqp  and all, 

I saw several similar issues during Windows installation in IPEX Git Hub. Like:
https://github.com/intel/intel-extension-for-pytorch/issues/485
https://github.com/intel/intel-extension-for-pytorch/issues/429 [Import intel_extension_for_pytorch fails on windows11

As we analyzed, the main root cause may still be in the installation environment, please note:


1. Neither “windows powershell” or “anaconda powershell prompt” does not work, if using conda prompt, we need to use "anaconda prompt".

please use the anaconda prompt itself, not in PS (powershell) environment. https://github.com/intel/intel-extension-for-pytorch/issues/485

2. Install step missing like missing install oneAPI or missing activate oneAPI, or missing other package:

conda install pkg-config libuv etc.

3. Other dependencies missing, like MSVC redistribution files.

so in summary, please create a new conda environment and
follow the install guide: https://intel.github.io/intel-extension-for-pytorch/index.html#installation?platform=gpu&version=v2.1.30%2bxpu&os=windows&package=pip

Pre-requisites installation **in order** :
3.1) GPU driver
3.2) MSVC environment or Install Microsoft Visual C++ Redistributable
Install Microsoft Visual C++ Redistributable from [Microsoft website](https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022).
3.3) Install oneAPI: 2024.1 {as no download link is available for 2023.2 anymore}
@SoldierWz if you haven't oneAPI install, you will run into the problem. please install oneAPI: [2024.1.0](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html)
3.4) open Python environment, like miconda3(latest) Python: 3.10

In anaconda prompt (please notes some VS code, call power shell by default)

3.5) Steps to reproduce:
conda create --name llmenv python=3.10
conda activate llm
call "C:\Program Files (x86)\Intel\oneAPI\compiler\latest\env\vars.bat" -> DPCPPROOT
call "C:\Program Files (x86)\Intel\oneAPI\compiler\mkl\env\vars.bat" -> MKLROOT

conda install pkg-config libuv
python -m pip install torch==2.1.0.post2 torchvision==0.16.0.post2 torchaudio==2.1.0.post2 intel-extension-for-pytorch==2.1.30 --extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/
Sanity test:
python -c "import torch; import intel_extension_for_pytorch as ipex; print(torch.version); print(ipex.version); [print(f'[{i}]: {torch.xpu.get_device_properties(i)}') for i in range(torch.xpu.device_count())];"

If you followed those and still run into the problem, please let us know the installation steps you are using. and collect your envs by this [collect_env.py](https://github.com/intel/intel-extension-for-pytorch/blob/main/scripts/collect_env.py)

Thanks

**Comment by yinghu5** [comment]
please feel free to reopen if any new issues
